### PR TITLE
feat: resolve commit hash

### PR DESCRIPTION
**Pull request by ondrej-hosak**

### CODE REVIEW

**Comment by ondrej-hosak** [comment]
@lksv Hi Lukas, please take a look at this pull request that adds resolving commit hash to package provider if you can spare few minutes.

File: lib/package_provider/cached_package.rb [hunk 1]
```diff
@@ -14,52 +14,41 @@ class PackingInProgress < StandardError
...
+      def package_path(package_hash)
+        File.join(PackageProvider.config.package_cache_root, package_hash)
       end
 
-      def errors(package_request_or_fingerprint)
-        fp = package_request_fingerprint(package_request_or_fingerprint)
-        return unless File.exist?(package_path(fp) + ERROR)
-        File.read(package_path(fp) + ERROR)
+      def errors(package_hash)
+        return unless File.exist?(package_path(package_hash) + ERROR)
```
Comment (Trawais) [inline lib/package_provider/cached_package.rb]: return nil unless...

File: lib/package_provider/cached_package.rb [hunk 2]
```diff
@@ -14,52 +14,41 @@ class PackingInProgress < StandardError
...
+        path_to_package(package_hash)
       end
 
-      def package_ready?(package_request_or_fingerprint)
-        path = package_path(package_request_or_fingerprint)
+      def package_ready?(package_hash)
+        path = package_path(package_hash)
 
         Dir.exist?(path) && File.exist?(path + PACKAGE_READY) &&
           File.exist?(File.join(path, 'package.zip')) &&
```
Comment (Trawais) [inline lib/package_provider/cached_package.rb]: Use method 'path_to_package' instead

File: lib/package_provider/cached_package.rb [hunk 3]
```diff
@@ -14,52 +14,41 @@ class PackingInProgress < StandardError
...
-        File.join(package_path(fp), 'package.zip')
-      end
-
-      def package_request_fingerprint(package_request_or_fingerprint)
-        if package_request_or_fingerprint.respond_to?(:fingerprint)
-          package_request_or_fingerprint.fingerprint
-        else
-          package_request_or_fingerprint
-        end
+      def path_to_package(package_hash)
```
Comment (Trawais) [inline lib/package_provider/cached_package.rb]: Confusing method names for 'package_path' and 'path_to_package' methods. I recommend something like 'package_directory' and 'package_fullpath'.

File: lib/package_provider/cached_package.rb [hunk 4]
```diff
@@ -14,52 +14,41 @@ class PackingInProgress < StandardError
...
+        File.join(package_path(package_hash), 'package.zip')
       end
     end
 
     attr_reader :package_request
 
-    def initialize(package_request)
+    def initialize(package_request, package_path)
```
Comment (Trawais) [inline lib/package_provider/cached_package.rb]: package_hash instead of package_path

File: lib/package_provider/repository_request.rb [hunk 5]
```diff
@@ -147,19 +148,16 @@ def to_json(options = {})
...
+      repo_valid? && revision_valid? && folder_override_valid?
     end
 
     def errors
       errors = []
-      errors << 'Repository is missing' unless repo
-      errors << 'Commit hash is missing' unless commit_hash
+      errors << 'Repository is missing' unless repo_valid?
+      errors << 'Commit hash and branch is missing' unless revision_valid?
       errors << folder_override.each_with_object([]) do |fo, s|
```
Comment (ireznice) [inline lib/package_provider/repository_request.rb]: I would recommend to change name of "s" variable

File: lib/package_provider/repository_request.rb [hunk 6]
```diff
@@ -112,6 +112,7 @@ def from_json(json)
     end
```
Comment (ireznice) [inline lib/package_provider/repository_request.rb]: the whole "sub-class" FolderOverride should be in different file

Comment (ondrej-hosak) [inline lib/package_provider/repository_request.rb]: I will do that in separate pull request.

**Pull request merged**